src/app/staticanalysis/modules/functionCallGraph.js

```diff
@@ -16,6 +16,30 @@ function buildLocalFuncCallGraphInternal (functions, nodeFilter, extractNodeIden
   return callGraph
 }
 
+/**
+ * Builds a function call graph for the current contracts.
+ * Example Contract call graph:
+ *
+ * {
+ *  "KingOfTheEtherThrone": {
+ *    "contracts": {...},                                        // Contract node as defined in abstractAstView.js
+ *    "functions": {
+ *      "KingOfTheEtherThrone.claimThrone(string memory)": {    // function in KingOfEtherThrone
```

Comment:
Oh I'm sorry, I did not see the comment.
Every source file is a scope. You cannot have two contracts of the same name in the same scope, but you can have them in different scopes. Example:

file1:

`contract c{}`

file 2:

`import "file1" as x; contract c{}`

The import is only required for remix so that it also pulls the other file into the compiler. If you only use `import "file1";`, the import will create an identifier clash in the scope of file2, but if you add `as x`, this can be avoided.

The sources are not in source order because there is no real source order and there is no "current file". What the compiler does in this case is that it tries to order the sources according to imports, but that also regularly fails. Can you do such an analysis in your part, too?